client/src/components/WorkoutSession/hooks/useActiveExercise.tsx

```diff
@@ -0,0 +1,51 @@
+import { ChangeEvent, InputHTMLAttributes, useState } from "react";
+import { useSetRecoilState } from "recoil";
+import { ID } from "../../../types/shared/id.types";
+import { ExerciseScheme } from "../../../types/shared/session.types";
+import { activeWorkoutState } from "../state/workout-state";
+import { isScheme } from "../types/is-scheme";
+
+export default function useActiveExercise(
```

Comment:
`useActiveExercise` _only_ does things related to warmup sets. Consider extracting the warmup form to a subcomponent, and renaming the hook to `useWarmupForm` or something similar.